api_server.go

```diff
@@ -278,13 +280,138 @@ func (x *ApiServer) SheetsUpload(w http.ResponseWriter, r *http.Request) {
 		}
 
 	default:
-		http.Error(w, "", http.StatusMethodNotAllowed)
+		methodNotAllowed(w)
+	}
+}
+
+type UploadType string
+
+func (x UploadType) String() string { return string(x) }
+
+const (
+	UploadTypeClickTrack UploadType = "click_track"
+	UploadTypeSheetMusic UploadType = "sheet_music"
+)
+
+type Upload struct {
+	UploadType        `json:"upload_type"`
+	*ClickTrackUpload `json:"click_track_upload"`
+	*SheetMusicUpload `json:"sheet_music_upload"`
+	FileName          string `json:"file_name"`
+	FileBytes         []byte `json:"file_bytes"`
+}
+
+type UploadStatus struct {
+	FileName string
+	Code     int
+	Error    string
+}
+
+type ClickTrackUpload struct {
+	ContentType string   `json:"content_type"`
+	Project     string   `json:"project"`
+	PartNames   []string `json:"part_names"`
+	PartNumbers []int    `json:"part_numbers"`
+}
+
+type SheetMusicUpload struct {
+	ContentType string   `json:"content_type"`
+	Project     string   `json:"project"`
```

Comment:
Is every upload associated with a project? Probably. So we might want to put this field in the `Upload` struct.